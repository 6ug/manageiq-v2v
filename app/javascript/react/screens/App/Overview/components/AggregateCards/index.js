import PendingTransformationPlans from './PendingTransformationPlans/PendingTransformationPlans';
import MigrationsInProgress from './MigrationsInProgress/MigrationsInProgress';
import MigrationsComplete from './MigrationsComplete/MigrationsComplete';
import InfrastructureMappings from './InfrastructureMappings/InfrastructureMappings';

export {
  PendingTransformationPlans,
  MigrationsInProgress,
  MigrationsComplete,
  InfrastructureMappings
};
