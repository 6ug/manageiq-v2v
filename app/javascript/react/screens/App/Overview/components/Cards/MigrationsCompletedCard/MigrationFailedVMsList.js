import React from 'react';
import PropTypes from 'prop-types';

import { Grid, OverlayTrigger, Tooltip } from 'patternfly-react';

const MigrationFailedVMsList = ({ migration }) => {
  const failedMigrations = migration.miq_request_tasks.filter(
    task => task.status === 'Error'
  );

  const getVMNameFromDescription = description => {
    const start = description.indexOf('[');
    const end = description.indexOf(']');

    if (start >= 0 && start < end) {
      return description.slice(start + 1, end);
    }

    return description;
  };

  return (
    <React.Fragment>
      <Grid fluid>
        <Grid.Row>
          <Grid.Col xs={4} md={3}>
            <strong>{__('VM Name')}</strong>
          </Grid.Col>
          <Grid.Col xs={8} md={9}>
            <strong>{__('Error Message')}</strong>
          </Grid.Col>
        </Grid.Row>
      </Grid>
      <Grid fluid className="failures-list">
        {failedMigrations.map(failure => (
          <Grid.Row key={failure.id}>
            <OverlayTrigger
              overlay={
                <Tooltip id={`failed-vm-name-${failure.id}`}>
                  {getVMNameFromDescription(failure.description)}
                </Tooltip>
              }
              placement="top"
              trigger={['hover']}
              delay={500}
              rootClose={false}
            >
              <Grid.Col className="failed-vm-column-text" xs={4} md={3}>
                {getVMNameFromDescription(failure.description)}
              </Grid.Col>
            </OverlayTrigger>
            <OverlayTrigger
              overlay={
                <Tooltip id={`failed-vm-name-${failure.id}`}>
                  {failure.message}
                </Tooltip>
              }
              placement="top"
              trigger={['hover']}
              delay={500}
              rootClose={false}
            >
              <Grid.Col className="failed-vm-column-text" xs={8} md={9}>
                {failure.message}
              </Grid.Col>
            </OverlayTrigger>
          </Grid.Row>
        ))}
      </Grid>
    </React.Fragment>
  );
};

MigrationFailedVMsList.propTypes = {
  migration: PropTypes.object
};
MigrationFailedVMsList.defaultProps = {
  migration: {}
};

export default MigrationFailedVMsList;
