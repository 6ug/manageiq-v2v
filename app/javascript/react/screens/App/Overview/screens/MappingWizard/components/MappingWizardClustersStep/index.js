import { connect } from 'react-redux';
import MappingWizardClustersStep from './MappingWizardClustersStep';
import * as MappingWizardClustersStepActions from './MappingWizardClustersStepActions';

import reducer from './MappingWizardClustersStepReducer';

export const reducers = { mappingWizardClustersStep: reducer };

const mapStateToProps = ({ mappingWizardClustersStep, form }, ownProps) => ({
  ...mappingWizardClustersStep,
  ...ownProps.data,
  targetProvider: form.mappingWizardGeneralStep.values.targetProvider
});

const mergeProps = (stateProps, dispatchProps, ownProps) => Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  MappingWizardClustersStepActions,
  mergeProps
)(MappingWizardClustersStep);
