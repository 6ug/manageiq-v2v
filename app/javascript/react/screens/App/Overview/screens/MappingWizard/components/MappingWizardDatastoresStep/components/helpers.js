import numeral from 'numeral';

const calculateTotalUsedSpace = sourceDatastores =>
  sourceDatastores.reduce(
    (totalSpace, sourceDatastore) =>
      (totalSpace += sourceDatastore.total_space - sourceDatastore.free_space),
    0
  );

const diskSpaceInfo = (targetDatastore, sourceDatastores) => {
  const { total_space, free_space } = targetDatastore;
  const availableSpace = free_space - calculateTotalUsedSpace(sourceDatastores);

  return sprintf(
    __('%s (%s MB total, %s avail)'),
    targetDatastore.name,
    numeral(total_space).format('0.00b'),
    numeral(availableSpace).format('0.00b')
  );
};

export { diskSpaceInfo };
