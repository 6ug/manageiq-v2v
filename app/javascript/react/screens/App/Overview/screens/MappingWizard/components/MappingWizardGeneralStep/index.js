import { connect } from 'react-redux';
import MappingWizardGeneralStep from './MappingWizardGeneralStep';
import { showAlertAction, hideAlertAction } from '../../MappingWizardActions';

const mapStateToProps = ({ overview: { transformationMappings }, form: { mappingWizardGeneralStep } }) => ({
  transformationMappings,
  targetProvider:
    mappingWizardGeneralStep && mappingWizardGeneralStep.values && mappingWizardGeneralStep.values.targetProvider
});

const actions = { showAlertAction, hideAlertAction };

export default connect(
  mapStateToProps,
  actions
)(MappingWizardGeneralStep);
