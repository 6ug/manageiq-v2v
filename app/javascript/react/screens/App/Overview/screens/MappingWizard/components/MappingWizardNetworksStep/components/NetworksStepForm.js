import React from 'react';
import PropTypes from 'prop-types';
import { bindMethods } from 'patternfly-react';
import cx from 'classnames';

import DualPaneMapper from '../../DualPaneMapper/DualPaneMapper';
import DualPaneMapperList from '../../DualPaneMapper/DualPaneMapperList';
import DualPaneMapperCount from '../../DualPaneMapper/DualPaneMapperCount';
import DualPaneMapperListItem from '../../DualPaneMapper/DualPaneMapperListItem';
import MappingWizardTreeView from '../../MappingWizardTreeView/MappingWizardTreeView';

import { sourceNetworksFilter } from '../MappingWizardNetworksStepSelectors';

class NetworksStepForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedSourceNetworks: [],
      selectedTargetNetwork: null,
      selectedNode: null
    };

    bindMethods(this, [
      'selectSourceNetwork',
      'selectTargetNetwork',
      'addNetworkMapping',
      'selectNode',
      'removeNode',
      'removeAll'
    ]);
  }

  componentWillReceiveProps(nextProps) {
    if (
      this.props.selectedCluster &&
      nextProps.selectedCluster.id !== this.props.selectedCluster.id
    ) {
      this.setState(() => ({
        selectedSourceNetworks: [],
        selectedTargetNetwork: null
      }));
    }
  }

  selectSourceNetwork(sourceNetwork) {
    this.setState(prevState => {
      const isAlreadySelected = prevState.selectedSourceNetworks.some(
        selectedSourceNetwork => selectedSourceNetwork.id === sourceNetwork.id
      );
      if (isAlreadySelected) {
        return {
          selectedSourceNetworks: prevState.selectedSourceNetworks.filter(
            selectedSourceNetwork =>
              selectedSourceNetwork.id !== sourceNetwork.id
          )
        };
      }
      return {
        selectedSourceNetworks: [
          ...prevState.selectedSourceNetworks,
          sourceNetwork
        ]
      };
    });
  }

  selectTargetNetwork(targetNetwork) {
    this.setState(() => ({ selectedTargetNetwork: targetNetwork }));
  }

  addNetworkMapping() {
    const { input, selectedCluster, selectedClusterMapping } = this.props;

    this.setState(prevState => {
      const networksStepMappings = input.value;

      const noMappingForTargetCluster = !networksStepMappings.some(
        targetClusterWithNetworkMappings =>
          // targetClusterWithNetworkMappings: { ...targetCluster, nodes: [...networkMappings] }
          // selectedClusterMapping: { ...targetCluster, nodes: [...sourceClusters (one of which is currently selected)]}
          targetClusterWithNetworkMappings.id === selectedClusterMapping.id
      );

      const addingToExistingMapping = networksStepMappings.some(
        targetClusterWithNetworkMappings =>
          targetClusterWithNetworkMappings.nodes.some(
            networkMapping =>
              // networkMapping: { ...targetNetwork, nodes: [...sourceNetworks] }
              networkMapping.id === prevState.selectedTargetNetwork.id
          )
      );

      if (networksStepMappings.length === 0 || noMappingForTargetCluster) {
        // There are no mappings, or there does not already exist a node for
        // a mapped target cluster
        input.onChange([
          ...networksStepMappings,
          {
            ...selectedClusterMapping,
            text: selectedClusterMapping.name,
            selectable: false,
            nodes: [
              {
                ...prevState.selectedTargetNetwork,
                text: prevState.selectedTargetNetwork.name,
                selectable: true,
                selected: false,
                state: {
                  expanded: true
                },
                nodes: prevState.selectedSourceNetworks.map(network => ({
                  ...network,
                  text: network.name,
                  icon: 'fa fa-file-o',
                  sourceClusterId: selectedCluster.id,
                  selectable: true,
                  selected: false
                }))
              }
            ]
          }
        ]);
      } else {
        input.onChange(
          networksStepMappings.map(targetClusterWithNetworkMappings => {
            // We are not adding a network mapping to this target cluster
            if (
              targetClusterWithNetworkMappings.id !== selectedClusterMapping.id
            ) {
              return targetClusterWithNetworkMappings;
              // There already exists a mapping for the currently selected target
              // network and we need to append the currently selected source networks
              // to its source networks array
            } else if (addingToExistingMapping) {
              return {
                ...targetClusterWithNetworkMappings,
                nodes: targetClusterWithNetworkMappings.nodes.map(
                  networkMapping => {
                    // networkMapping: { ...targetNetwork, nodes: [...sourceNetworks] }
                    // If the target network for this mapping matches the currently
                    // selected target network, add the currently selected source
                    // networks to its source networks array
                    if (
                      networkMapping.id === prevState.selectedTargetNetwork.id
                    ) {
                      return {
                        ...networkMapping,
                        nodes: networkMapping.nodes.concat(
                          prevState.selectedSourceNetworks.map(network => ({
                            ...network,
                            text: network.name,
                            icon: 'fa fa-file-o',
                            sourceClusterId: selectedCluster.id,
                            selectable: true,
                            selected: false
                          }))
                        )
                      };
                    }
                    return networkMapping;
                  }
                )
              };
            }
            // We are creating a new mapping to add under the target cluster
            return {
              ...targetClusterWithNetworkMappings,
              nodes: targetClusterWithNetworkMappings.nodes.concat({
                ...prevState.selectedTargetNetwork,
                text: prevState.selectedTargetNetwork.name,
                selectable: true,
                selected: false,
                state: {
                  expanded: true
                },
                nodes: prevState.selectedSourceNetworks.map(network => ({
                  ...network,
                  text: network.name,
                  icon: 'fa fa-file-o',
                  sourceClusterId: selectedCluster.id,
                  selectable: true,
                  selected: false
                }))
              })
            };
          })
        );
      }
      return {
        selectedTargetNetwork: null,
        selectedSourceNetworks: []
      };
    });
  }

  selectNode(selectedNode) {
    const { value: networksStepMappings, onChange } = this.props.input;
    const isTargetNetwork = selectedNode.nodes;

    if (isTargetNetwork) {
      const updatedMappings = networksStepMappings.map(
        targetClusterWithNetworksMappings => {
          const {
            nodes: networksMappings,
            ...targetCluster
          } = targetClusterWithNetworksMappings;
          return {
            ...targetCluster,
            nodes: networksMappings.map(networksMapping => {
              const {
                nodes: sourceNetworks,
                ...targetNetwork
              } = networksMapping;
              return targetNetwork.id === selectedNode.id
                ? {
                    ...targetNetwork,
                    selected: !targetNetwork.selected,
                    nodes: sourceNetworks.map(sourceNetwork => ({
                      ...sourceNetwork,
                      selected: false
                    }))
                  }
                : {
                    ...targetNetwork,
                    selected: false,
                    nodes: sourceNetworks.map(sourceNetwork => ({
                      ...sourceNetwork,
                      selected: false
                    }))
                  };
            })
          };
        }
      );
      onChange(updatedMappings);
    } else {
      const updatedMappings = networksStepMappings.map(
        targetClusterWithNetworksMappings => {
          const {
            nodes: networksMappings,
            ...targetCluster
          } = targetClusterWithNetworksMappings;
          return {
            ...targetCluster,
            nodes: networksMappings.map(networksMapping => {
              const {
                nodes: sourceNetworks,
                ...targetNetwork
              } = networksMapping;
              return {
                ...targetNetwork,
                selected: false,
                nodes: sourceNetworks.map(sourceNetwork => {
                  if (sourceNetwork.id === selectedNode.id) {
                    return {
                      ...sourceNetwork,
                      selected: !sourceNetwork.selected
                    };
                  } else if (sourceNetwork.selected) {
                    return {
                      ...sourceNetwork,
                      selected: false
                    };
                  }
                  return sourceNetwork;
                })
              };
            })
          };
        }
      );
      onChange(updatedMappings);
    }
    this.setState(() => ({ selectedNode }));
  }

  removeNode() {
    const { value: networksStepMappings, onChange } = this.props.input;
    const { selectedNode } = this.state;
    const isTargetNetwork = selectedNode.nodes;

    if (isTargetNetwork) {
      const updatedMappings = networksStepMappings
        .map(targetClusterWithNetworksMappings => {
          const {
            nodes: networksMappings,
            ...targetCluster
          } = targetClusterWithNetworksMappings;
          const updatedNetworksMappings = networksMappings.filter(
            targetNetworkWithSourceNetworks =>
              targetNetworkWithSourceNetworks.id !== selectedNode.id
          );
          return updatedNetworksMappings.length === 0
            ? undefined
            : {
                ...targetCluster,
                nodes: updatedNetworksMappings
              };
        })
        .filter(item => item !== undefined);
      onChange(updatedMappings);
    } else {
      const updatedMappings = networksStepMappings
        .map(targetClusterWithNetworksMappings => {
          const {
            nodes: networksMappings,
            ...targetCluster
          } = targetClusterWithNetworksMappings;
          const updatedNetworksMappings = networksMappings
            .map(networksMapping => {
              const {
                nodes: sourceNetworks,
                ...targetNetwork
              } = networksMapping;
              const updatedSourceNetworks = sourceNetworks.filter(
                sourceNetwork => sourceNetwork.id !== selectedNode.id
              );
              return updatedSourceNetworks.length === 0
                ? undefined
                : {
                    ...targetNetwork,
                    nodes: updatedSourceNetworks
                  };
            })
            .filter(item => item !== undefined);
          return updatedNetworksMappings.length === 0
            ? undefined
            : {
                ...targetCluster,
                nodes: updatedNetworksMappings
              };
        })
        .filter(item => item !== undefined);
      onChange(updatedMappings);
    }
    this.setState(() => ({ selectedNode: null }));
  }

  removeAll() {
    const { input } = this.props;
    input.onChange([]);
  }

  render() {
    const {
      sourceNetworks,
      targetNetworks,
      isFetchingSourceNetworks,
      isFetchingTargetNetworks,
      input,
      selectedCluster
    } = this.props;
    const {
      selectedSourceNetworks,
      selectedTargetNetwork,
      selectedNode
    } = this.state;

    const classes = cx('dual-pane-mapper-form', {
      'is-hidden': !selectedCluster
    });

    return (
      <div className={classes}>
        <DualPaneMapper
          handleButtonClick={this.addNetworkMapping}
          validMapping={
            !(
              selectedTargetNetwork &&
              (selectedSourceNetworks && selectedSourceNetworks.length > 0)
            )
          }
        >
          <DualPaneMapperList
            listTitle="Source Networks"
            loading={isFetchingSourceNetworks}
          >
            {sourceNetworks &&
              sourceNetworksFilter(sourceNetworks, input.value).map(
                sourceNetwork => (
                  <DualPaneMapperListItem
                    item={sourceNetwork}
                    text={sourceNetwork.name}
                    key={sourceNetwork.id}
                    selected={
                      selectedSourceNetworks &&
                      selectedSourceNetworks.some(
                        selectedSourceNetwork =>
                          selectedSourceNetwork.id === sourceNetwork.id
                      )
                    }
                    handleClick={this.selectSourceNetwork}
                    handleKeyPress={this.selectSourceNetwork}
                  />
                )
              )}
            <DualPaneMapperCount
              selectedItems={selectedSourceNetworks.length}
              totalItems={
                sourceNetworksFilter(sourceNetworks, input.value).length
              }
            />
          </DualPaneMapperList>
          <DualPaneMapperList
            listTitle="Target Networks"
            loading={isFetchingTargetNetworks}
          >
            {targetNetworks &&
              targetNetworks.map(targetNetwork => (
                <DualPaneMapperListItem
                  item={targetNetwork}
                  text={targetNetwork.name}
                  key={targetNetwork.id}
                  selected={
                    selectedTargetNetwork &&
                    selectedTargetNetwork.id === targetNetwork.id
                  }
                  handleClick={this.selectTargetNetwork}
                  handleKeyPress={this.selectTargetNetwork}
                />
              ))}
          </DualPaneMapperList>
        </DualPaneMapper>
        <MappingWizardTreeView
          mappings={input.value}
          selectNode={this.selectNode}
          removeNode={this.removeNode}
          removeAll={this.removeAll}
          selectedNode={selectedNode}
        />
      </div>
    );
  }
}

NetworksStepForm.propTypes = {
  input: PropTypes.object,
  sourceNetworks: PropTypes.array,
  targetNetworks: PropTypes.array,
  isFetchingSourceNetworks: PropTypes.bool,
  isFetchingTargetNetworks: PropTypes.bool,
  selectedCluster: PropTypes.object,
  selectedClusterMapping: PropTypes.object
};

export default NetworksStepForm;
