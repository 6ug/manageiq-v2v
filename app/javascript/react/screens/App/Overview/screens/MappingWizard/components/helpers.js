import { V2V_TARGET_PROVIDERS } from '../MappingWizardConstants';

export const getClusterOptions = clusterMappings => {
  const sourceClustersWithAssociatedTargetClusters = clusterMappings.reduce(
    (mappings, targetClusterWithSourceClusters) => {
      const { nodes: sourceClusters, ...targetCluster } = targetClusterWithSourceClusters;
      const sourceToTargetMappings = sourceClusters.map(sourceCluster => ({
        sourceCluster,
        targetCluster,
        sourceClusterMappedToTargetCluster: {
          name: `${sourceCluster.name} (${targetCluster.name})`,
          id: sourceCluster.id
        }
      }));
      return mappings.concat(sourceToTargetMappings);
    },
    []
  );

  return sourceClustersWithAssociatedTargetClusters.map(
    ({ sourceClusterMappedToTargetCluster }) => sourceClusterMappedToTargetCluster
  );
};

export const multiProviderTargetLabel = (providerType, wizardStep) => {
  switch (providerType) {
    case V2V_TARGET_PROVIDERS[1].id:
      switch (wizardStep) {
        case 'cluster':
          return __('Target Provider \\ Project');
        case 'network':
          return __('Target Project \\ Network');
        default:
          return null;
      }
    default:
      switch (wizardStep) {
        case 'cluster':
          return __('Target Provider \\ Datacenter \\ Cluster');
        case 'network':
          return __('Target Networks');
        default:
          return null;
      }
  }
};
