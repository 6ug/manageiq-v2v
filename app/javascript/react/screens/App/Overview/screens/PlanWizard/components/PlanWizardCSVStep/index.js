import { connect } from 'react-redux';
import PlanWizardCSVStep from './PlanWizardCSVStep';

export default connect()(PlanWizardCSVStep);
