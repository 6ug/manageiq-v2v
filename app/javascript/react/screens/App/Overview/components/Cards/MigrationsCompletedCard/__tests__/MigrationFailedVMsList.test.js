import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MigrationFailedVMsList from '../MigrationFailedVMsList';
import { coreComponents } from '../../../../../../../../components/index';
import componentRegistry from '../../../../../../../../components/componentRegistry';
import { requestCompletedServiceRequests } from '../migrationsCompleted.fixtures';

jest.mock('../../../../../../../../components/componentRegistry');

componentRegistry.registerMultiple(coreComponents);

describe('MigrationFailedVMsList component', () => {
  const migrations = requestCompletedServiceRequests.response.data.resources;

  it('renders the MigrationsCompletedCard', () => {
    const component = shallow(
      <MigrationFailedVMsList migration={migrations[1]} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
