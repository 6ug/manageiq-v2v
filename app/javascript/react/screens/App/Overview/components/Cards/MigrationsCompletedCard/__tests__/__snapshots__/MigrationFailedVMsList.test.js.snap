// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MigrationFailedVMsList component renders the MigrationsCompletedCard 1`] = `
<React.Fragment>
  <Grid
    fluid={true}
  >
    <Row>
      <Col
        md={3}
        xs={4}
      >
        <strong>
          VM Name
        </strong>
      </Col>
      <Col
        md={9}
        xs={8}
      >
        <strong>
          Error Message
        </strong>
      </Col>
    </Row>
  </Grid>
  <Grid
    className="failures-list"
    fluid={true}
  >
    <Row
      key="24"
    >
      <OverlayTrigger
        defaultOverlayShown={false}
        delay={500}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="failed-vm-name-24"
            placement="right"
          >
            vmmigrate
          </Tooltip>
        }
        placement="top"
        rootClose={false}
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <Col
          className="failed-vm-column-text"
          md={3}
          xs={4}
        >
          vmmigrate
        </Col>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        delay={500}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="failed-vm-name-24"
            placement="right"
          >
            Service_Template_Provisioning failed
          </Tooltip>
        }
        placement="top"
        rootClose={false}
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <Col
          className="failed-vm-column-text"
          md={9}
          xs={8}
        >
          Service_Template_Provisioning failed
        </Col>
      </OverlayTrigger>
    </Row>
    <Row
      key="25"
    >
      <OverlayTrigger
        defaultOverlayShown={false}
        delay={500}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="failed-vm-name-25"
            placement="right"
          >
            vmmigrate
          </Tooltip>
        }
        placement="top"
        rootClose={false}
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <Col
          className="failed-vm-column-text"
          md={3}
          xs={4}
        >
          vmmigrate
        </Col>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        delay={500}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="failed-vm-name-25"
            placement="right"
          >
            Service_Template_Provisioning failed
          </Tooltip>
        }
        placement="top"
        rootClose={false}
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <Col
          className="failed-vm-column-text"
          md={9}
          xs={8}
        >
          Service_Template_Provisioning failed
        </Col>
      </OverlayTrigger>
    </Row>
    <Row
      key="26"
    >
      <OverlayTrigger
        defaultOverlayShown={false}
        delay={500}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="failed-vm-name-26"
            placement="right"
          >
            vmmigrate
          </Tooltip>
        }
        placement="top"
        rootClose={false}
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <Col
          className="failed-vm-column-text"
          md={3}
          xs={4}
        >
          vmmigrate
        </Col>
      </OverlayTrigger>
      <OverlayTrigger
        defaultOverlayShown={false}
        delay={500}
        overlay={
          <Tooltip
            bsClass="tooltip"
            id="failed-vm-name-26"
            placement="right"
          >
            Service_Template_Provisioning failed
          </Tooltip>
        }
        placement="top"
        rootClose={false}
        trigger={
          Array [
            "hover",
          ]
        }
      >
        <Col
          className="failed-vm-column-text"
          md={9}
          xs={8}
        >
          Service_Template_Provisioning failed
        </Col>
      </OverlayTrigger>
    </Row>
  </Grid>
</React.Fragment>
`;
